plot_cfDNA_scATAC/scATAC_demo.Rmd

```diff
@@ -0,0 +1,430 @@
+pbmc <- RenameIdents(
+  object = pbmc,
+  '0' = 'CD14 Mono',
+  '1' = 'CD4 Memory',
+  '2' = 'CD8 Effector',
+  '3' = 'CD4 Naive',
+  '4' = 'CD14 Mono',
+  '5' = 'DN T',
+  '6' = 'CD8 Naive',
+  '7' = 'NK CD56Dim',
+  '8' = 'pre-B',
+  '9' = 'CD16 Mono',
+  '10' = 'pro-B',
+  '11' = 'DC',
+  '12' = 'NK CD56bright',
+  '13' = 'pDC'
+)
+pbmc <- RenameIdents(
+  object = pbmc,
+  'CD14 Mono' = 'Mono',
+  'CD4 Memory' = 'T-cells',
+  'CD8 Effector' = 'T-cells',
+  'CD4 Naive' = 'T-cells',
+  'CD14 Mono' = 'Mono',
+  'DN T' = 'T-cells',
+  'CD8 Naive' = 'T-cells',
+  'NK CD56Dim' = 'NK',
+  'pre-B' = 'B-cells',
+  'CD16 Mono' = 'Mono',
+  'pro-B' = 'pro-B',
+  'DC' = 'DC',
+  'NK CD56bright' = 'NK',
+  'pDC' = 'DC'
+)
+pbmc <- RenameIdents(
+  object = pbmc,
+  'Mono' = 'Other',
+  'T-cells' = 'Other',
+  'T-cells' = 'Other',
+  'T-cells' = 'Other',
+  'Mono' = 'Other',
+  'T-cells' = 'Other',
+  'T-cells' = 'Other',
+  'NK' = 'Other',
+  'B-cells' = 'B-cells',
+  'Mono' = 'Other',
+  'pro-B' = 'B-cells',
+  'DC' = 'Other',
+  'NK' = 'Other',
+  'DC' = 'Other'
+)
+
+```
+
+
+# Get the TFBS motifs and footprints
+
+We will compare these with the footprints gathers from cfDNA
+
+```{r tfbs_setup, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+pbmc_rna =NA
+gene.activities = NA
+counts = NA
+
+DefaultAssay(pbmc) <- 'peaks'
+
+pwm <- getMatrixSet(
+  x = JASPAR2020,
+  opts = list(species = 9606, all_versions = FALSE)
+)
+
+all_motifs = lapply(pwm, function(x) name(x))
+motif_idx = grep("EBF1", unlist(all_motifs))
+pwm_select = pwm[motif_idx]
+    
+pbmc <- AddMotifs(pbmc, genome = BSgenome.Hsapiens.UCSC.hg19, pfm = pwm_select)
+
+```
+
+### Plot footprints
+
+```{r tfbs_analysis, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# gather the footprinting information for sets of motifs
+
+# gather the footprinting information for sets of motifs
+pbmc2 <- Footprint(
+  object = subset(x = pbmc, idents = c("B-cells", "Other")),
+  motif.name = c("EBF1"),
+  genome = BSgenome.Hsapiens.UCSC.hg19,upstream = 300, downstream = 300,in.peaks = F
+)
+footprint_df = GetFootprintData(pbmc2, features = c("EBF1"))
+
+# plot the footprint data for each group of cells
+p2 <- PlotFootprint(pbmc2, features = c("EBF1"))
+p2
+pdf("data/data_alexis/atac_plot.pdf", height=2, width=4)
+print(p2)
+dev.off()
+```
+
+### plot smoothed footprints
+
+```{r cubic_spline, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# gather the footprinting information for sets of motifs
+
+library(ggplot2)
+theme_set(theme_bw())
+library(signal)
+library(zoo)
+
+gg_footprint = ggplot(footprint_df, aes(x=position, y=norm.value, color=group)) + 
+                #geom_line() +
+                geom_line(aes(y=rollmean(norm.value, 100, na.pad=TRUE))) +
+                theme_bw() + theme(legend.position = "none", aspect.ratio = 9/16) +
+                labs(x ="Position Along TFBS", y = "Normalized Count")
+gg_footprint
```

Comment:
This could use a legend for what the colors mean.

Reply:
ah, yes, added here: https://github.com/greenelab/cfDNA_atac_demo/pull/3/commits/e1b7e639f576324543ec6a2d47756582a56f2758